workload/{bank,insights}: tests non-deterministically fail to split ranges when pointed to a secondary tenant

**Describe the problem**

after #108762, the tests in pkg/workload/bank and /insights fail non-terministically (e.g. under stress) when pointed to a secondary tenant:

```
=== RUN   TestInsightsWorkload/rows=10/ranges=9
    insights_test.go:74: got 1 ranges expected 9
=== RUN   TestInsightsWorkload/rows=10/ranges=10
    insights_test.go:74: got 1 ranges expected 10
=== RUN   TestInsightsWorkload/rows=10/ranges=100
    insights_test.go:74: got 1 ranges expected 10
=== CONT  TestInsightsWorkload
    insights_test.go:78: -- test log scope end --
test logs left over in: /tmp/_tmp/eba056330b249b393168d116b1921cec/logTestInsightsWorkload3333307126
--- FAIL: TestInsightsWorkload (2.85s)
    --- PASS: TestInsightsWorkload/rows=10/ranges=0 (0.07s)
    --- PASS: TestInsightsWorkload/rows=10/ranges=1 (0.20s)
    --- FAIL: TestInsightsWorkload/rows=10/ranges=9 (0.19s)
    --- FAIL: TestInsightsWorkload/rows=10/ranges=10 (0.19s)
    --- FAIL: TestInsightsWorkload/rows=10/ranges=100 (0.19s)
```

Same error for:
- `TestBankWorkload` in workload/bank
- `TestSplits` in workloadsql

**Expected behavior**

The test should deterministically succeed.

Epic: CRDB-26687


Jira issue: CRDB-30935

--- a/pkg/workload/bank/bank_test.go
+++ b/pkg/workload/bank/bank_test.go
@@ -42,11 +42,7 @@ func TestBank(t *testing.T) {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
-	srv, db, _ := serverutils.StartServer(t, base.TestServerArgs{
-		UseDatabase: `test`,
-
-		DefaultTestTenant: base.TestIsForStuffThatShouldWorkWithSecondaryTenantsButDoesntYet(109458),
-	})
+	srv, db, _ := serverutils.StartServer(t, base.TestServerArgs{UseDatabase: `test`})
 	defer srv.Stopper().Stop(ctx)
 
 	sqlutils.MakeSQLRunner(db).Exec(t, `CREATE DATABASE test`)

--- a/pkg/workload/insights/insights_test.go
+++ b/pkg/workload/insights/insights_test.go
@@ -43,11 +43,7 @@ func TestInsightsWorkload(t *testing.T) {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
-	srv, db, _ := serverutils.StartServer(t, base.TestServerArgs{
-		UseDatabase: `test`,
-
-		DefaultTestTenant: base.TestIsForStuffThatShouldWorkWithSecondaryTenantsButDoesntYet(109458),
-	})
+	srv, db, _ := serverutils.StartServer(t, base.TestServerArgs{UseDatabase: `test`})
 	defer srv.Stopper().Stop(ctx)
 
 	sqlutils.MakeSQLRunner(db).Exec(t, `CREATE DATABASE test`)

--- a/pkg/workload/workloadsql/workloadsql_test.go
+++ b/pkg/workload/workloadsql/workloadsql_test.go
@@ -81,11 +81,7 @@ func TestSplits(t *testing.T) {
 	defer log.Scope(t).Close(t)
 
 	ctx := context.Background()
-	srv, db, _ := serverutils.StartServer(t, base.TestServerArgs{
-		UseDatabase: `test`,
-
-		DefaultTestTenant: base.TestIsForStuffThatShouldWorkWithSecondaryTenantsButDoesntYet(109458),
-	})
+	srv, db, _ := serverutils.StartServer(t, base.TestServerArgs{UseDatabase: `test`})
 	defer srv.Stopper().Stop(ctx)
 
 	sqlDB := sqlutils.MakeSQLRunner(db)

